pre_processing/read_slstr_funcs.F90

```diff
@@ -229,13 +233,31 @@ subroutine read_slstr_visdata(indir, inband, outarr, imager_angles, &
    ! Find the filename required for this channel
    call get_slstr_imnames(indir, inband, filename, filename_qa, bandname, irradname)
 
-   allocate(data1(nx*2,ny*2))
+   ! Open the netcdf file
+   call ncdf_open(fid, filename, 'read_slstr_visdata()')
+   ! Determin the actual size of the data array (which should be
+   ! nx*2,ny*2, but sometimes isn't). Ensure the data array is at
+   ! least nx*2, ny*2 in size
+   ny1 = ncdf_dim_length(fid, 'rows', 'read_slstr_visdata()')
+   nx1 = ncdf_dim_length(fid, 'columns', 'read_slstr_visdata()')
+   if (ny1 .gt. ny*2) then
+      nyr = ny1
+   else
+      nyr = ny*2
+   end if
+   if (nx1 .gt. nx*2) then
+      nxr = nx1
+   else
+      nxr = nx*2
+   end if
+
+   allocate(data1(nxr,nyr))
```

Comment:
I don't think this bit does anything unless there can be more than 1 extra row/column, the possible states are `ny1 == ny*2` and `ny1 == ny*2+1`, which means `nyr == ny1` in both cases.

Reply:
Fixed